File: demo-service-impl/src/main/java/com/github/zubmike/service/demo/dao/StarshipDao.java
package com.github.zubmike.service.demo.dao;

import com.github.zubmike.service.dao.EntityItemDao;
import com.github.zubmike.service.demo.types.Starship;

import java.util.Optional;

public interface StarshipDao extends EntityItemDao<Long, Starship> {

	Optional<Starship> getByNumber(String number);

}

File: demo-service-kotlin-impl/src/main/kotlin/com/github/zubmike/service/demo/dao/StarshipDao.kt
package com.github.zubmike.service.demo.dao

import com.github.zubmike.core.dao.EntityItemDao
import com.github.zubmike.service.demo.types.Starship

interface StarshipDao : EntityItemDao<Long, Starship> {

    fun getByNumber(number: String): Starship?

}
